apps/studio/components/interfaces/Organization/BillingSettings/Subscription/SubscriptionPlanUpdateDialog.tsx

```diff
@@ -0,0 +1,644 @@
+import { useQueryClient } from '@tanstack/react-query'
+import { InfoIcon, Check, ExternalLink } from 'lucide-react'
+import Link from 'next/link'
+import { useState, useEffect } from 'react'
+import { toast } from 'sonner'
+import tweets from 'shared-data/tweets'
+
+import AlertError from 'components/ui/AlertError'
+import ShimmeringLoader from 'components/ui/ShimmeringLoader'
+import {
+  Table,
+  TableBody,
+  TableCell,
+  TableHead,
+  TableHeader,
+  TableRow,
+  HoverCard,
+  HoverCardContent,
+  HoverCardTrigger,
+  Card,
+  CardContent,
+  Badge,
+} from 'ui'
+import { useOrgSubscriptionUpdateMutation } from 'data/subscriptions/org-subscription-update-mutation'
+import { organizationKeys } from 'data/organizations/keys'
+import { formatCurrency } from 'lib/helpers'
+import { PRICING_TIER_PRODUCT_IDS } from 'lib/constants'
+import { SubscriptionTier } from 'data/subscriptions/types'
+import { billingPartnerLabel } from 'components/interfaces/Billing/Subscription/Subscription.utils'
+import PaymentMethodSelection from './PaymentMethodSelection'
+import { Button, Dialog, DialogContent } from 'ui'
+import { InfoTooltip } from 'ui-patterns/info-tooltip'
+import { OrganizationBillingSubscriptionPreviewResponse } from 'data/organizations/organization-billing-subscription-preview'
+
+const getRandomTweet = () => {
```

Comment:
lets dedupe with the random tweet logic in SignInLayout too?
https://github.com/supabase/supabase/blob/feat/edge-functions-layout-single/apps/studio/components/layouts/SignInLayout/SignInLayout.tsx#L78